iceoryx_hoofs/container/include/iox/vector.hpp

```diff
@@ -43,96 +43,115 @@ class vector final
+    /// @deterministic
     vector& operator=(vector&& rhs) noexcept;
 
     /// @brief returns an iterator to the first element of the vector,
     ///         if the vector is empty it returns the same iterator as
     ///         end (the first iterator which is outside of the vector)
+    /// @deterministic
     iterator begin() noexcept;
 
     /// @brief returns a const iterator to the first element of the vector,
     ///         if the vector is empty it returns the same iterator as
     ///         end (the first iterator which is outside of the vector)
+    /// @deterministic
     const_iterator begin() const noexcept;
 
     /// @brief returns an iterator to the element which comes after the last
     ///         element (the first element which is outside of the vector)
+    /// @deterministic
     iterator end() noexcept;
 
     /// @brief returns a const iterator to the element which comes after the last
     ///         element (the first element which is outside of the vector)
+    /// @deterministic
     const_iterator end() const noexcept;
 
     /// @brief return the pointer to the underlying array
     /// @return pointer to underlying array
+    /// @deterministic
     T* data() noexcept;
 
     /// @brief return the const pointer to the underlying array
     /// @return const pointer to underlying array
+    /// @deterministic
     const T* data() const noexcept;
 
     /// @brief returns a reference to the element stored at index.
     /// @param[in] index of the element to return
     /// @return reference to the element stored at index
     /// @attention Out of bounds access leads to a program termination!
+    /// @deterministic
     T& at(const uint64_t index) noexcept;
 
     /// @brief returns a const reference to the element stored at index.
     /// @param[in] index of the element to return
     /// @return const reference to the element stored at index
     /// @attention Out of bounds access leads to a program termination!
+    /// @deterministic
     const T& at(const uint64_t index) const noexcept;
 
     /// @brief returns a reference to the element stored at index.
     /// @param[in] index of the element to return
     /// @return reference to the element stored at index
     /// @attention Out of bounds access leads to a program termination!
+    /// @deterministic
     T& operator[](const uint64_t index) noexcept;
 
     /// @brief returns a const reference to the element stored at index.
     /// @param[in] index of the element to return
     /// @return const reference to the element stored at index
     /// @attention Out of bounds access leads to a program termination!
+    /// @deterministic
     const T& operator[](const uint64_t index) const noexcept;
 
     /// @brief returns a reference to the first element; terminates if the vector is empty
     /// @return reference to the first element
     /// @attention Accessing an empty vector leads to a program termination!
+    /// @deterministic
```

Comment:
What about the const front method?

Reply:
Oversight. One really needs tooling for this e.g. based on LLVM.